ghost/members-api/lib/repositories/MemberRepository.js

```diff
@@ -281,7 +282,29 @@ module.exports = class MemberRepository {
             memberStatusData.status = 'comped';
         }
 
-        // Subscribe member to default newsletters
+        //checks for custom signUp forms
+        if (memberData.type === 'subscribe' && memberData.newsletters) {
+            const allValidNewsletters = await this._newslettersService.browse({limit: 'all'});
+            const allValidNewslettersIds = allValidNewsletters.map(newsletter => newsletter.id);
+
+            if (memberData.newsletters) {
+                const subscribedNewsletterIds = memberData.newsletters.map(newsletter => newsletter.id);
+                const invalidIds = subscribedNewsletterIds.filter(id => !allValidNewslettersIds.includes(id));
+                if (invalidIds.length > 0) {
+                    throw new errors.BadRequestError({message: tpl(messages.invalidNewsLetterId, {id: invalidIds})});
+                } else if (memberData.newsletters.length > 0) {
```

Comment:
I don't think we need the `else if` here - If `invalidIds.length > 0` then an error is thrown which short circuits the rest of the function

Reply:
Yes, makes sense!